ext/bcmath/libbcmath/src/recmul.c

```diff
@@ -90,6 +90,43 @@ static inline void bc_fast_square(bc_num n1, size_t n1len, bc_num *prod)
 	}
 }
 
+/** Common part of functions bc_standard_mul and bc_standard_square */
+static inline void bc_standard_mul_common(BC_VECTOR *prod_vector, size_t prod_arr_size, size_t prodlen, bc_num *prod, BC_VECTOR *buf) {
```

Comment:
```suggestion
/* Common part of functions bc_standard_mul and bc_standard_square that takes a vector and converts it to a bc_num */
static inline void bc_mul_finish_from_vector(BC_VECTOR *prod_vector, size_t prod_arr_size, size_t prodlen, bc_num *prod, BC_VECTOR *buf) {
```

I would also suggest to keep the `buf` cleanup for the caller to make the memory allocation/cleanup intention clear: generally the policy is that a function should perform its own cleanup.